src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -6196,11 +6170,10 @@ private TypeSyntax ParseTypeArgument()
             // or a,... which  means this case should happen less frequently than what we're 
             // trying to solve here so we err on the side of better error messages
             // for the majority of cases.
-            SyntaxKind nextTokenKind = SyntaxKind.None;
 
             if (result.IsMissing &&
                 (this.CurrentToken.Kind != SyntaxKind.CommaToken && this.CurrentToken.Kind != SyntaxKind.GreaterThanToken) &&
```

Comment:
```suggestion
                this.CurrentToken.Kind is not SyntaxKind.CommaToken and not SyntaxKind.GreaterThanToken &&
```

Reply:
Ah yeah. I was planning on doing a pass for the != Checks. I'll add that today.